docs/tutorials/python/mnist.md

```diff
@@ -1,116 +1,135 @@
 
 ```python
 import mxnet as mx
 mnist = mx.test_utils.get_mnist()
 ```
 
-Next we create data iterators for MXNet. A data iterator returns a batch of
-examples with according labels each time. If the examples are images, then they
-are represented by a 4-D matrix with shape `(batch_size, num_channels, width,
-height)`. For the MNIST dataset, there is only one color channel, and both width
-and height are 28, therefore the shape is `(batch_size, 1, 28, 28)`. In
-addition, we often shuffle the images used for training, which accelerates the
-training progress.
+After running the above source code, the entire MNIST dataset should be fully loaded into memory. Note that for large datasets it is not feasible to pre-load the entire dataset first like we did here. What is needed is a mechanism by which we can quickly and efficiently stream data directly from the source. MXNet Data iterators come to the rescue here by providing exactly that. Data iterator is the mechanism by which we feed input data into an MXNet training algorithm and they are very simple to initialize and use and are optimized for speed. During training, we typically process training samples in small batches and over the entire training lifetime will end up processing each training example multiple times. In this tutorial, we'll configure the data iterator to feed examples in batches of 100. Keep in mind that each example is a 28x28 grayscale image and the corresponding label.
+
+Image batches are commonly represented by a 4-D matrix with shape `(batch_size, num_channels, width, height)`. For MNIST dataset, since the images are grayscale, there is only one color channel. Also, the images are 28x28 pixels, and so each image has width and height equal to 28. Therefore, the shape of input is `(batch_size, 1, 28, 28)`. Another important consideration is the order of input samples. When feeding training examples, it is critical that we don't feed samples with the same label in succession. Doing so can slow down training.
+Data iterators take care of this by randomly shuffling the inputs. Note that we only need to shuffle train data. The order does not matter for test data.
+
+The following source code initializes the data iterators for MNIST dataset. Note that we initialize two iterators: one for train data and one for test data.
 
 ```python
 batch_size = 100
 train_iter = mx.io.NDArrayIter(mnist['train_data'], mnist['train_label'], batch_size, shuffle=True)
 val_iter = mx.io.NDArrayIter(mnist['test_data'], mnist['test_label'], batch_size)
 ```
 
+## Training
+We will cover a couple of approaches for performing the hand written digit recognition task. The first approach makes use of a more traditional deep neural network architecture called Multilayer Percepton. We'll discuss its drawbacks and use that as a motivation to introduce a second more advanced approach called Convolution Neural Network that has proven to work very well for image classification tasks.
```

Comment:
"more traditional" -> "traditional" (we can't compare it to anything because we haven't introduced another model yet)